lua/rocks/operations/add.lua

```diff
@@ -87,16 +88,16 @@ add.add = function(arg_list, callback, opts)
 
     nio.run(function()
         helpers.semaphore.with(function()
-            local user_rocks = helpers.parse_rocks_toml()
-            local handler = handlers.get_install_handler_callback(user_rocks, arg_list)
+            local user_rocks = helpers.parse_rocks_toml(opts.config_path)
+            local handler = handlers.get_install_handler_callback(user_rocks --[[@as MutRocksTomlRef]], arg_list)
             if type(handler) == "function" then
                 local function report_progress(message)
                     progress_handle:report({
                         message = message,
                     })
                 end
                 handler(report_progress, report_error, helpers.manage_rock_stub)
-                user_rocks:write()
```

Comment:
question: is this change necessary?
I find `user_rocks:write()` more readable than `user_rocks()`.

Reply:
It was for the purpose of removing the use of non-meta fields/methods from the `MultiMutRocksTomlWrapper` type so that it would be logically equivalent to the `MutRocksTomlRef` type. Right now, the two types are almost equivalent in that if you access an index from the wrapper object, you'll only get whatever is defined in the table object types. But there is an additional "write" field in the wrapper type which will be given preference over the same key in the tables (see the `__index` I added in the last PR). This could create issues in the future as there could be clashing field names between the config and the wrapper class. Since this is just meant to be helper class that should represent the config type, I didn't want there to be any consideration required or possible issues in the future when adding/renaming fields in the config file just for a helper class.

Reply:
I see. To me, it feels very unlikely that the table object type would have a field called `write`. But to be on the safer side, we could call the field something like `_write_await` (something highly unlikely to be used, and also indicating that it's internal with the `_` prefix and hinting the async context with the `_await` suffix).

We could then document in the api module that rocks.toml keys should not have  `_` prefixes, thus reserving them for potential meta  fields in the future.
Theoretically, this would be a breaking change, but I wouldn't treat it as one in this case, because I know for a fact that no rocks.nvim extension adds `_`-prefixed keys to the rocks.toml schema.

@vhyrro any thoughts on this?

Reply:
Yeah, I was also thinking about using the `_` prefix to denote that it's an internal method but I still wasn't sure if that would still pose a risk. I'm fine with whatever you guys decide!

Another thought could be that the `__call` metamethod could accept a function of type `fun(configs: MutRocksTomlRefWithPath[])` and pass in the internal config table to allow then method you pass in to act on them as individual config tables. Then you could have an external function (as part of the `multi_mut_rocks_toml_wrapper` module) called something like `write_rocks_toml` which would take in a wrapper object and pass a function to it which writes each config object to their respective paths?

Reply:
Done